### DISCUSSION TITLE
Deepgram produced WebVTT files can be missing the milliseconds component of timestamps

### DISCUSSION THREAD

**Discussion by jjherman** [deepgram/community#786] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT

### Details

We noticed a failure to parse some of the captions in a Deepgram produced WebVTT file. Digging into the file, we found that the Deepgram STT API appears to sometimes generate timestamps missing the milliseconds component (see the start time of cue 80 in the screenshot).

![image](https://github.com/deepgram/community/assets/2013615/34a564a4-557c-437b-a4cc-8d0fea22b107)

Per the [WebVTT timestamp specification](https://www.w3.org/TR/webvtt1/#webvtt-timestamp), the milliseconds component of the timestamp is not optional, so I believe that it should be written to the VTT file (possibly as `.000` to denote zero milliseconds).

### If possible, please attach your code or paste it into the text box.

We're just using the Deepgram SDK to call the API; the problem is occurring on Deepgram's back end.

### If possible, please attach an example audio file to reproduce the issue.

Note, this file is a M4A, but has been renamed to MP4 to get around GitHub's restrictions on file uploads.
https://github.com/deepgram/community/assets/2013615/47182f0f-a52e-4492-8a5c-739c032351c6

That said, this file has been truncated from the original, so it may not exhibit the problem as I could not get a version of the original to upload to GitHub within the 10 MB limit.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment]
@jjherman  just to confirm you are using our Speech to Text  (STT) API here, not TTS as you mentioned in the question above.

**Reply by jjherman** [reply]
Correct @jpvajda; I have updated the text to reflect that I'm using Deepgram's Speech to Text API.

**Comment by jjherman** [comment, marked as answer]
My apologies @jpvajda , but it appears that the issue I have is actually in the .NET SDK and not the STT API. Since that has a different GitHub repository, I am going to close out this issue.
